src/btree/bt_read.c

```diff
@@ -265,18 +267,21 @@ __page_read(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags)
      * If the function building an in-memory version of the page failed, it discarded the page, but
      * not the disk image. Discard the page and separately discard the disk image in all cases.
      */
-    if (ref->page != NULL)
+    if (ref->page != NULL) {
+        disk_image_freed = ref->page->dsk != NULL;
+        __wt_page_modify_clear(session, ref->page);
```

Comment:
Why do we need this call?

Reply:
ref out will assert if we discard a dirty page. But here it is fine to discard.